src/project/project-actions.tsx

```diff
@@ -106,9 +108,10 @@ export class ProjectActions {
       await this.dialogs.show<void>((callback) => (
         <WebUSBDialog callback={callback} action={WebUSBErrorTrigger.Connect} />
       ));
+      await this.download();
     } else {
       if (await this.showConnectHelp(forceConnectHelp)) {
-        await this.connectInternal();
+        return await this.connectInternal();
```

Comment:
Do we need this to be "return await" ? Normally only needed for the scope of try/catch which we don't have here.